Github pages action: fatal: not in a git directory

<!-- Please check the Q&A before posting an issue: https://github.com/JamesIves/github-pages-deploy-action/discussions?discussions_q=category%3AQ%26A -->

## Describe the bug

Github page action: Doesn't work..



## Reproduction Steps

<!-- Steps to reproduce the behavior. -->

## Logs

Run JamesIves/github-pages-deploy-action@4.2.0
/usr/bin/docker exec xxxx sh -c "cat /etc/*release | grep ^ID"

    GitHub Pages Deploy Action 🚀

    💖 Support: https://github.com/sponsors/JamesIves
    📣 Maintained by James Ives: https://jamesiv.es/

    🚀 Getting Started Guide: https://github.com/JamesIves/github-pages-deploy-action
    ❓ Discussions / Q&A: https://github.com/JamesIves/github-pages-deploy-action/discussions
    🔧 Report a Bug: https://github.com/JamesIves/github-pages-deploy-action/issues
Checking configuration and starting deployment… 🚦
The operating system you're using is not supported and results may be varied. Please refer to the documentation for more details. ❗
Deploying using Deploy Token… 🔑
Configuring git…
/usr/bin/git config user.name xxx
fatal: not in a git directory
Error: There was an error initializing the repository: The process '/usr/bin/git' failed with exit code 1[28] ❌
Deployment failed! ❌
## Workflow

- name: Deploy to Github Pages
        uses: JamesIves/github-pages-deploy-action@4.2.0
        with:
          branch: gh-pages
          folder: build/docs

at least for me there is not much more in the logs (even with `ACTIONS_STEP_DEBUG=true`, full log file attached below):

```
[...]
2022-04-22T07:49:44.9008714Z ##[debug]Starting: Deploy Development Documentation
2022-04-22T07:49:44.9041704Z ##[debug]Loading inputs
2022-04-22T07:49:44.9045405Z ##[debug]Evaluating: secrets.SSH_DEPLOY_KEY
2022-04-22T07:49:44.9045994Z ##[debug]Evaluating Index:
2022-04-22T07:49:44.9046506Z ##[debug]..Evaluating secrets:
2022-04-22T07:49:44.9047028Z ##[debug]..=> Object
2022-04-22T07:49:44.9047530Z ##[debug]..Evaluating String:
2022-04-22T07:49:44.9048030Z ##[debug]..=> 'SSH_DEPLOY_KEY'
2022-04-22T07:49:44.9049024Z ##[debug]=> '***
2022-04-22T07:49:44.9049743Z ##[debug]***
2022-04-22T07:49:44.9050436Z ##[debug]***
2022-04-22T07:49:44.9051096Z ##[debug]***
2022-04-22T07:49:44.9051733Z ##[debug]***
2022-04-22T07:49:44.9052395Z ##[debug]***
2022-04-22T07:49:44.9053042Z ##[debug]***
2022-04-22T07:49:44.9053714Z ##[debug]***
2022-04-22T07:49:44.9054333Z ##[debug]***
2022-04-22T07:49:44.9054952Z ##[debug]***
2022-04-22T07:49:44.9055557Z ##[debug]***
2022-04-22T07:49:44.9056259Z ##[debug]***
2022-04-22T07:49:44.9056944Z ##[debug]***
2022-04-22T07:49:44.9057613Z ##[debug]***
2022-04-22T07:49:44.9058242Z ##[debug]***
2022-04-22T07:49:44.9058911Z ##[debug]***
2022-04-22T07:49:44.9059531Z ##[debug]***
2022-04-22T07:49:44.9060214Z ##[debug]***
2022-04-22T07:49:44.9060811Z ##[debug]***
2022-04-22T07:49:44.9061694Z ##[debug]***
2022-04-22T07:49:44.9062329Z ##[debug]***
2022-04-22T07:49:44.9062992Z ##[debug]***
2022-04-22T07:49:44.9063683Z ##[debug]***
2022-04-22T07:49:44.9064320Z ##[debug]***
2022-04-22T07:49:44.9064990Z ##[debug]***
2022-04-22T07:49:44.9065634Z ##[debug]***
2022-04-22T07:49:44.9066314Z ##[debug]***
2022-04-22T07:49:44.9066923Z ##[debug]***
2022-04-22T07:49:44.9067607Z ##[debug]***
2022-04-22T07:49:44.9068222Z ##[debug]***
2022-04-22T07:49:44.9068902Z ##[debug]***
2022-04-22T07:49:44.9069597Z ##[debug]***
2022-04-22T07:49:44.9070276Z ##[debug]***
2022-04-22T07:49:44.9071127Z ##[debug]***
2022-04-22T07:49:44.9071817Z ##[debug]***
2022-04-22T07:49:44.9072496Z ##[debug]***
2022-04-22T07:49:44.9073183Z ##[debug]***
2022-04-22T07:49:44.9073853Z ##[debug]***
2022-04-22T07:49:44.9074503Z ##[debug]***
2022-04-22T07:49:44.9075145Z ##[debug]***
2022-04-22T07:49:44.9075794Z ##[debug]***
2022-04-22T07:49:44.9076486Z ##[debug]***
2022-04-22T07:49:44.9077242Z ##[debug]***
2022-04-22T07:49:44.9077850Z ##[debug]***
2022-04-22T07:49:44.9078551Z ##[debug]***
2022-04-22T07:49:44.9079214Z ##[debug]***
2022-04-22T07:49:44.9079902Z ##[debug]***
2022-04-22T07:49:44.9080556Z ##[debug]***
2022-04-22T07:49:44.9081064Z ##[debug]***'
2022-04-22T07:49:44.9081843Z ##[debug]Result: '***
2022-04-22T07:49:44.9082559Z ##[debug]***
2022-04-22T07:49:44.9083212Z ##[debug]***
2022-04-22T07:49:44.9083812Z ##[debug]***
2022-04-22T07:49:44.9084443Z ##[debug]***
2022-04-22T07:49:44.9085108Z ##[debug]***
2022-04-22T07:49:44.9085754Z ##[debug]***
2022-04-22T07:49:44.9086426Z ##[debug]***
2022-04-22T07:49:44.9087054Z ##[debug]***
2022-04-22T07:49:44.9087669Z ##[debug]***
2022-04-22T07:49:44.9088288Z ##[debug]***
2022-04-22T07:49:44.9088952Z ##[debug]***
2022-04-22T07:49:44.9089633Z ##[debug]***
2022-04-22T07:49:44.9090297Z ##[debug]***
2022-04-22T07:49:44.9090958Z ##[debug]***
2022-04-22T07:49:44.9091592Z ##[debug]***
2022-04-22T07:49:44.9092211Z ##[debug]***
2022-04-22T07:49:44.9092891Z ##[debug]***
2022-04-22T07:49:44.9093499Z ##[debug]***
2022-04-22T07:49:44.9094106Z ##[debug]***
2022-04-22T07:49:44.9094729Z ##[debug]***
2022-04-22T07:49:44.9095395Z ##[debug]***
2022-04-22T07:49:44.9096065Z ##[debug]***
2022-04-22T07:49:44.9096673Z ##[debug]***
2022-04-22T07:49:44.9097330Z ##[debug]***
2022-04-22T07:49:44.9097958Z ##[debug]***
2022-04-22T07:49:44.9098629Z ##[debug]***
2022-04-22T07:49:44.9099230Z ##[debug]***
2022-04-22T07:49:44.9099897Z ##[debug]***
2022-04-22T07:49:44.9100505Z ##[debug]***
2022-04-22T07:49:44.9101281Z ##[debug]***
2022-04-22T07:49:44.9101981Z ##[debug]***
2022-04-22T07:49:44.9102655Z ##[debug]***
2022-04-22T07:49:44.9103303Z ##[debug]***
2022-04-22T07:49:44.9103978Z ##[debug]***
2022-04-22T07:49:44.9104642Z ##[debug]***
2022-04-22T07:49:44.9105317Z ##[debug]***
2022-04-22T07:49:44.9105970Z ##[debug]***
2022-04-22T07:49:44.9106601Z ##[debug]***
2022-04-22T07:49:44.9107226Z ##[debug]***
2022-04-22T07:49:44.9107864Z ##[debug]***
2022-04-22T07:49:44.9108547Z ##[debug]***
2022-04-22T07:49:44.9109207Z ##[debug]***
2022-04-22T07:49:44.9109797Z ##[debug]***
2022-04-22T07:49:44.9110483Z ##[debug]***
2022-04-22T07:49:44.9111158Z ##[debug]***
2022-04-22T07:49:44.9111812Z ##[debug]***
2022-04-22T07:49:44.9112465Z ##[debug]***
2022-04-22T07:49:44.9112957Z ##[debug]***'
2022-04-22T07:49:44.9113978Z ##[debug]Evaluating: github.token
2022-04-22T07:49:44.9114512Z ##[debug]Evaluating Index:
2022-04-22T07:49:44.9115009Z ##[debug]..Evaluating github:
2022-04-22T07:49:44.9115540Z ##[debug]..=> Object
2022-04-22T07:49:44.9116028Z ##[debug]..Evaluating String:
2022-04-22T07:49:44.9116536Z ##[debug]..=> 'token'
2022-04-22T07:49:44.9117307Z ##[debug]=> '***'
2022-04-22T07:49:44.9117930Z ##[debug]Result: '***'
2022-04-22T07:49:44.9123221Z ##[debug]Loading env
2022-04-22T07:49:44.9130847Z ##[group]Run JamesIves/github-pages-deploy-action@releases/v4
2022-04-22T07:49:44.9131442Z with:
2022-04-22T07:49:44.9131909Z   branch: gh-pages
2022-04-22T07:49:44.9132384Z   folder: build/doc
2022-04-22T07:49:44.9132889Z   target-folder: develop
2022-04-22T07:49:44.9133368Z   clean: true
2022-04-22T07:49:44.9134441Z   clean-exclude: releases/

2022-04-22T07:49:44.9147558Z   ssh-key: ***
2022-04-22T07:49:44.9148418Z   token: ***
2022-04-22T07:49:44.9149113Z   force: true
2022-04-22T07:49:44.9158706Z ##[endgroup]
2022-04-22T07:49:44.9177097Z ##[command]/usr/bin/docker exec  70afe1b9a4ef88f84d78b2f4e1b8e8412a3ec086338d5414d4864942440c4f31 sh -c "cat /etc/*release | grep ^ID"
2022-04-22T07:49:45.0205408Z ##[debug]ID=ubuntu
2022-04-22T07:49:45.0205926Z ##[debug]ID_LIKE=debian
2022-04-22T07:49:45.0206503Z ##[debug]Running JavaScript Action with default external tool: node12
2022-04-22T07:49:45.2219403Z 
2022-04-22T07:49:45.2220423Z     ╭━━━╮╭╮╭╮╱╭╮╱╱╭╮╱╱╭━━━╮
2022-04-22T07:49:45.2221366Z     ┃╭━╮┣╯╰┫┃╱┃┃╱╱┃┃╱╱┃╭━╮┃
2022-04-22T07:49:45.2221962Z     ┃┃╱╰╋╮╭┫╰━╯┣╮╭┫╰━╮┃╰━╯┣━━┳━━┳━━┳━━╮
2022-04-22T07:49:45.2222579Z     ┃┃╭━╋┫┃┃╭━╮┃┃┃┃╭╮┃┃╭━━┫╭╮┃╭╮┃┃━┫━━┫
2022-04-22T07:49:45.2223179Z     ┃╰┻━┃┃╰┫┃╱┃┃╰╯┃╰╯┃┃┃╱╱┃╭╮┃╰╯┃┃━╋━━┃
2022-04-22T07:49:45.2228864Z     ╰━━━┻┻━┻╯╱╰┻━━┻━━╯╰╯╱╱╰╯╰┻━╮┣━━┻━━╯
2022-04-22T07:49:45.2229540Z     ╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╭━╯┃
2022-04-22T07:49:45.2230116Z     ╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╰━━╯
2022-04-22T07:49:45.2230680Z     ╭━━━╮╱╱╱╱╱╭╮╱╱╱╱╱╱╱╱╭━━━╮╱╱╭╮
2022-04-22T07:49:45.2231231Z     ╰╮╭╮┃╱╱╱╱╱┃┃╱╱╱╱╱╱╱╱┃╭━╮┃╱╭╯╰╮
2022-04-22T07:49:45.2231825Z     ╱┃┃┃┣━━┳━━┫┃╭━━┳╮╱╭╮┃┃╱┃┣━┻╮╭╋┳━━┳━╮
2022-04-22T07:49:45.2232426Z     ╱┃┃┃┃┃━┫╭╮┃┃┃╭╮┃┃╱┃┃┃╰━╯┃╭━┫┃┣┫╭╮┃╭╮╮
2022-04-22T07:49:45.2233041Z     ╭╯╰╯┃┃━┫╰╯┃╰┫╰╯┃╰━╯┃┃╭━╮┃╰━┫╰┫┃╰╯┃┃┃┃
2022-04-22T07:49:45.2233639Z     ╰━━━┻━━┫╭━┻━┻━━┻━╮╭╯╰╯╱╰┻━━┻━┻┻━━┻╯╰╯
2022-04-22T07:49:45.2234288Z     ╱╱╱╱╱╱╱┃┃╱╱╱╱╱╱╭━╯┃
2022-04-22T07:49:45.2234806Z     ╱╱╱╱╱╱╱╰╯╱╱╱╱╱╱╰━━╯
2022-04-22T07:49:45.2235260Z     
2022-04-22T07:49:45.2235502Z 
2022-04-22T07:49:45.2235907Z     💖 Support: https://github.com/sponsors/JamesIves
2022-04-22T07:49:45.2236607Z     📣 Maintained by James Ives: https://jamesiv.es
2022-04-22T07:49:45.2236959Z 
2022-04-22T07:49:45.2237458Z     🚀 Getting Started Guide: https://github.com/JamesIves/github-pages-deploy-action
2022-04-22T07:49:45.2238495Z     ❓ Discussions / Q&A: https://github.com/JamesIves/github-pages-deploy-action/discussions
2022-04-22T07:49:45.2239444Z     🔧 Report a Bug: https://github.com/JamesIves/github-pages-deploy-action/issues
2022-04-22T07:49:45.2240642Z Checking configuration and starting deployment… 🚦
2022-04-22T07:49:45.2247994Z Configuring SSH client… 🔑
2022-04-22T07:49:45.2427646Z Identity added: (stdin) (tiziano.mueller@chem.uzh.ch)
2022-04-22T07:49:45.2477553Z Deploying using SSH Deploy Key… 🔑
2022-04-22T07:49:45.2481346Z Configuring git…
2022-04-22T07:49:45.2504350Z [command]/usr/bin/git config user.name dev-zero
2022-04-22T07:49:45.2545020Z fatal: not in a git directory
2022-04-22T07:49:45.2589816Z ##[error]There was an error initializing the repository: The process '/usr/bin/git' failed with exit code 128 ❌
2022-04-22T07:49:45.2600603Z ##[notice]Deployment failed! ❌
2022-04-22T07:49:45.2601516Z 
2022-04-22T07:49:45.2602605Z ::set-output name=deployment-status::failed
2022-04-22T07:49:45.2605469Z ##[debug]='failed'
2022-04-22T07:49:45.2691401Z ##[debug]Node Action run completed with exit code 1
2022-04-22T07:49:45.2704719Z ##[debug]SSH_AUTH_SOCK='/tmp/ssh-Bl4lC1Ij5PAr/agent.1652'
2022-04-22T07:49:45.2705483Z ##[debug]SSH_AGENT_PID='1653'
2022-04-22T07:49:45.2706031Z ##[debug]deployment_status='failed'
2022-04-22T07:49:45.2710558Z ##[debug]Finishing: Deploy Development Documentation
[...]
```

[dbcsr-doc-deploy-log.txt](https://github.com/JamesIves/github-pages-deploy-action/files/8538380/dbcsr-doc-deploy-log.txt)

--- a/__tests__/git.test.ts
+++ b/__tests__/git.test.ts
@@ -55,7 +55,7 @@ describe('git', () => {
       })
 
       await init(action)
-      expect(execute).toBeCalledTimes(6)
+      expect(execute).toBeCalledTimes(7)
     })
 
     it('should catch when a function throws an error', async () => {
@@ -102,7 +102,7 @@ describe('git', () => {
       })
 
       await init(action)
-      expect(execute).toBeCalledTimes(6)
+      expect(execute).toBeCalledTimes(7)
     })
 
     it('should not unset git config if a user is using ssh', async () => {
@@ -124,7 +124,7 @@ describe('git', () => {
       })
 
       await init(action)
-      expect(execute).toBeCalledTimes(5)
+      expect(execute).toBeCalledTimes(6)
 
       process.env.CI = undefined
     })
@@ -145,7 +145,7 @@ describe('git', () => {
       })
 
       await init(action)
-      expect(execute).toBeCalledTimes(6)
+      expect(execute).toBeCalledTimes(7)
     })
   })
 

--- a/__tests__/main.test.ts
+++ b/__tests__/main.test.ts
@@ -53,7 +53,7 @@ describe('main', () => {
       debug: true
     })
     await run(action)
-    expect(execute).toBeCalledTimes(16)
+    expect(execute).toBeCalledTimes(17)
     expect(rmRF).toBeCalledTimes(1)
     expect(exportVariable).toBeCalledTimes(1)
   })
@@ -73,7 +73,7 @@ describe('main', () => {
       isTest: TestFlag.HAS_CHANGED_FILES
     })
     await run(action)
-    expect(execute).toBeCalledTimes(19)
+    expect(execute).toBeCalledTimes(20)
     expect(rmRF).toBeCalledTimes(1)
     expect(exportVariable).toBeCalledTimes(1)
   })

--- a/src/git.ts
+++ b/src/git.ts
@@ -22,6 +22,12 @@ export async function init(action: ActionInterface): Promise<void | Error> {
     info('Configuring git…')
 
     await execute(
+      `git config --global --add safe.directory "${action.workspace}"`,
+      action.workspace,
+      action.silent
+    )
+
+    await execute(
       `git config user.name "${action.name}"`,
       action.workspace,
       action.silent

--- a/src/git.ts
+++ b/src/git.ts
@@ -21,11 +21,17 @@ export async function init(action: ActionInterface): Promise<void | Error> {
     info(`Deploying using ${action.tokenType}… 🔑`)
     info('Configuring git…')
 
-    await execute(
-      `git config --global --add safe.directory "${action.workspace}"`,
-      action.workspace,
-      action.silent
-    )
+    try {
+      await execute(
+        `git config --global --add safe.directory "${action.workspace}"`,
+        action.workspace,
+        action.silent
+      )
+    } catch {
+      info(
+        'Unable to set workspace as a safe directory…'
+      )
+    }
 
     await execute(
       `git config user.name "${action.name}"`,

--- a/src/git.ts
+++ b/src/git.ts
@@ -28,9 +28,7 @@ export async function init(action: ActionInterface): Promise<void | Error> {
         action.silent
       )
     } catch {
-      info(
-        'Unable to set workspace as a safe directory…'
-      )
+      info('Unable to set workspace as a safe directory…')
     }
 
     await execute(

